[PM-17562] Update for new event integrations configurations (#575)
bitwarden/contributing-docs: docs/getting-started/server/events.md

        "password": "SET_A_PASSWORD_HERE_123",
        "exchangeName": "events-exchange",
        "eventRepositoryQueueName": "events-write-queue",
        "slackQueueName": "events-slack-queue",
        "webhookQueueName": "events-webhook-queue",
      }
    }
    ```

>     :::info (optional) The `slackQueueName` and `webhookQueueName` specified above are optional. If

Reviewer: I think it's because the `:::` isn't on a new line.

1. Add the following to your `secrets.json` in `dev` to configure the service bus:

   ```json
       "eventLogging": {
         "azureServiceBus": {
           "connectionString": "\"Endpoint=sb://localhost;SharedAccessKeyName=RootManageSharedAccessKey;SharedAccessKey=SAS_KEY_VALUE;UseDevelopmentEmulator=true;\"",
           "topicName": "event-logging",
           "eventRepositorySubscriptionName": "events-write-subscription",
           "slackSubscriptionName": "events-slack-subscription",
           "webhookSubscriptionName": "events-webhook-subscription",
         }
       },
   ```

>    :::info (optional) The `slackSubscriptionName` and `webhookSubscriptionName` specified above are

Reviewer: Same issue as above too.

3. Start or re-start all services, including `EventsProcessor`.

> ### Integrations and IntegrationConfigurations

Reviewer: Alternatively, we make this "... and integration configurations".

> ### Integrations and IntegrationConfigurations

Author: I like making these simple words - "integration configurations" - since we're not using the full class names from code and we're just referring to the concepts.

    ```json
    { "token": "xoxb-token-from-slack" }
    ```

  - For webhooks, it is `null`. However, even though there is no configuration, an organization must
    have a webhook `OrganizationIntegration` to enable configuration via
    `OrganizationIntegrationConfiguration`

**OrganizationIntegrationConfiguration**

- This contains the configurations specific to each `EventType` for the integration.
- `Configuration` contains the event-specific configuration

  - For Slack, this would contain what channel to send the message to:

    ```json
    { "channelId": "C123456" }
    ```

  - For Webhook, this is the URL the request should be sent to:

    ```json
    { "url": "https://api.example.com" }
    ```

- `Template` contains a template string the is expected to be filled in with the contents of the
  actual event.
  - The tokens in the string are wrapped in `#` characters. For instance, the UserId would be
    `#UserId#`
  - The `IntegrationTemplateProcessor` does the actual work of replacing these tokens with
    introspected values from the provided `EventMessage`.
  - Note that the template does not enforce any structure - it could be a freeform text message to

>     send via Slack, or a JSON body to send via webhook. We simply store and use it as a string for

Reviewer: ```suggestion
    send via Slack, or a JSON body to send via webhook; it is simply stored and used as a string for
```

    the most flexibility.

> **OrganizationIntegrationConfigurationDetails**

Reviewer: ```suggestion
**`OrganizationIntegrationConfigurationDetails`**
```

Reviewer: I think you can drop this whole section actually -- we don't really document these and I think its purpose is clear.
